common/src/main/java/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -409,6 +415,57 @@ class Backend(
         }
     }
 
+    fun getProductEntitlementMappings(
+        onSuccessHandler: (ProductEntitlementMappings) -> Unit,
+        onErrorHandler: (PurchasesError) -> Unit
+    ) {
+        val endpoint = Endpoint.GetProductEntitlementMappings
+        val path = endpoint.getPath()
+        val call = object : Dispatcher.AsyncCall() {
+            override fun call(): HTTPResult {
+                return httpClient.performRequest(
+                    appConfig.baseURL,
+                    endpoint,
+                    null,
+                    authenticationHeaders
+                )
+            }
+
+            override fun onError(error: PurchasesError) {
+                synchronized(this@Backend) {
+                    productEntitlementCallbacks.remove(path)
+                }?.forEach { (_, onError) ->
+                    onError(error)
+                }
+            }
+
+            override fun onCompletion(result: HTTPResult) {
+                synchronized(this@Backend) {
+                    productEntitlementCallbacks.remove(path)
+                }?.forEach { (onSuccess, onError) ->
+                    if (result.isSuccessful()) {
+                        try {
+                            onSuccess(ProductEntitlementMappings.fromJson(result.body))
+                        } catch (e: JSONException) {
+                            onError(e.toPurchasesError().also { errorLog(it) })
+                        }
+                    } else {
+                        onError(result.toPurchasesError().also { errorLog(it) })
+                    }
+                }
+            }
+        }
+        synchronized(this@Backend) {
+            productEntitlementCallbacks.addCallback(
+                call,
+                dispatcher,
+                path,
+                onSuccessHandler to onErrorHandler,
+                Delay.DEFAULT
+            )
+        }
+    }
```

Comment:
There is a bit of repetition on this file. It might be good to start refactoring this class to avoid that... But just adding this for now.